### PR TITLE
NEXUS-41151: Upgrade to JUnit5

### PR DESCRIPTION
**Pull request by gshockv**
Ticket: [NEXUS-41151](https://sonatype.atlassian.net/browse/NEXUS-41151)
CI: 
- [Goodies Feature](https://jenkins.ci.sonatype.dev/job/nxrm/job/libraries/job/goodies-feature/job/NEXUS-41151-junit5/)
- [NXRM PR Builder](https://jenkins.ci.sonatype.dev/job/nxrm/job/nxrm3/job/PR_Builder/job/NEXUS-41151-goodies-junit5/)

[NEXUS-41151]: https://sonatype.atlassian.net/browse/NEXUS-41151?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: httpfixture/pom.xml [hunk 1]
```diff
@@ -66,9 +66,8 @@
     </dependency>
 
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <optional>true</optional>
+      <groupId>org.junit.vintage</groupId>
+      <artifactId>junit-vintage-engine</artifactId>
```
Comment (bobotimi) [inline httpfixture/pom.xml:70]: Does this dependency allow us to write both JUnit 4 and JUnit 5 tests?

Comment (gshockv) [reply httpfixture/pom.xml:70]: Added dependency for `junit-jupiter`. It allows (in combination with `junit-vingate`) to write both junit4/junit5 tests.

Comment (gshockv) [reply httpfixture/pom.xml:70]: https://github.com/sonatype/goodies/pull/97/commits/a34412f256b6290ffb80a7bfecc128c06951b3b4

**Review by jbermudezsonatype — APPROVED** [review]

**Review by bobotimi — APPROVED** [review]
I think it'll be a good idea to do a build of nexus-internal on a branch which points to this new version of the goodies dependency

**Comment by gshockv** [comment]
> I think it'll be a good idea to do a build of nexus-internal on a branch which points to this new version of the goodies dependency

[NXRM PR Builder](https://jenkins.ci.sonatype.dev/job/nxrm/job/nxrm3/job/PR_Builder/job/NEXUS-41151-goodies-junit5/)

**Pull request merged**